### PR TITLE
Add sharing argument and deprecate world_readable. Fixes #332

### LINKED ISSUE #332: `secret` sharing option
In the Plotly Python library, we replaced the `world_readable` argument with the `privacy` argument which allows for a third sharing option: "secret". This works by making a second `PATCH` request: https://github.com/plotly/plotly.py/blob/master/plotly/plotly/plotly.py#L1328-L1365

We should add this to the R library as well.

### PR DESCRIPTION
**Pull request by cpsievert**

### CODE REVIEW

File: R/plotly_POST.R [hunk 1]
```diff
@@ -55,6 +72,14 @@ plotly_POST <- function(x, filename, fileopt = "new", world_readable = TRUE) {
   base_url <- file.path(get_domain(), "clientresp")
   resp <- httr::POST(base_url, body = bod)
   con <- process(struct(resp, "clientresp"))
+  if (sharing[1] == "secret") {
+    bits <- strsplit(con$url, "/")[[1]]
+    plot_id <- bits[length(bits)]
+    url <- paste0(get_domain("v2"), "files/", verify("username"), ":", plot_id)
+    bod <- list(share_key_enabled = TRUE)
+    con2 <- httr::PATCH(url, plotly_headers("v2"), body = bod, encode = "json")
+    con$url <- paste0(con$url, "?share_key=", content(con2)$share_key)
```
Comment (cpsievert) [inline R/plotly_POST.R]: @chriddyp is there anything else from the `PATCH` response that you'd like to return?

**Comment by cpsievert** [comment]
> The message below was automatically generated after build https://travis-ci.org/ropensci/plotly/builds/103256104 

 Detected 3 differences -> 

 http://cpsievert.github.io/plotly-test-table/8e3d3bb/jitter-basic
http://cpsievert.github.io/plotly-test-table/8e3d3bb/scatterplots-geom_jitter
http://cpsievert.github.io/plotly-test-table/8e3d3bb/scatterplots-jitter

**Comment by cpsievert** [comment]
> The message below was automatically generated after build https://travis-ci.org/ropensci/plotly/builds/103256354 

 Detected 3 differences -> 

 http://cpsievert.github.io/plotly-test-table/75b576c/jitter-basic
http://cpsievert.github.io/plotly-test-table/75b576c/scatterplots-geom_jitter
http://cpsievert.github.io/plotly-test-table/75b576c/scatterplots-jitter

File: R/plotly_POST.R [hunk 2]
```diff
@@ -34,12 +46,17 @@ plotly_POST <- function(x, filename, fileopt = "new", world_readable = TRUE) {
...
+    warning("fileopt was specified in the wrong place.",
+            "Please specify in plotly_POST()")
+  }
   x$fileopt <- fileopt
-  if (!is.null(x$world_readable)) 
-    warning("world_readable was specified in the wrong place. Please specify in plotly_POST()")
-  x$world_readable <- world_readable
+  if (!is.null(x$world_readable)) {
+    warning("world_readable was specified in the wrong place.",
+            "Please use the sharing argument in plotly_POST()")
```
Comment (chriddyp) [inline R/plotly_POST.R]: Maybe instead:

> world_readable is no longer supported. Instead, set the "sharing" argument to "private" (you must be logged in to access), "secret" (anybody with the obscured URL can access) or "public" (anybody can view).

File: R/plotly_POST.R (hunk 1 above)
Comment (chriddyp) [inline R/plotly_POST.R]: I don't think so!

**Comment by chriddyp** [comment]
looks good to me! @delekru and I have been trying to come up with better names for "secret". Some options include: "hidden" or "unlisted". I think "unlisted" explains the behaviour the best. Thoughts?

**Comment by cpsievert** [comment]
Thanks! No strong preference. I think I prefer 'unlisted'...how about 'key'?

**Comment by chriddyp** [comment]
But my friend prefers "hidden", so I'm open.

I think the "key" thing is confusing to folks that aren't developers

**Comment by cpsievert** [comment]
OK, go ahead with 'hidden'?

**Comment by cpsievert** [comment]
> The message below was automatically generated after build https://travis-ci.org/ropensci/plotly/builds/103264405 

 Detected 3 differences -> 

 http://cpsievert.github.io/plotly-test-table/82b8462/jitter-basic
http://cpsievert.github.io/plotly-test-table/82b8462/scatterplots-geom_jitter
http://cpsievert.github.io/plotly-test-table/82b8462/scatterplots-jitter

**Comment by cpsievert** [comment]
> The message below was automatically generated after build https://travis-ci.org/ropensci/plotly/builds/103372409 

 Detected 3 differences -> 

 http://cpsievert.github.io/plotly-test-table/619f9fb/jitter-basic
http://cpsievert.github.io/plotly-test-table/619f9fb/scatterplots-geom_jitter
http://cpsievert.github.io/plotly-test-table/619f9fb/scatterplots-jitter

**Pull request merged**